### ISSUE TITLE
Moneymanagerex crashes when deleting a transaction

### ISSUE THREAD

**Issue by johnnyab47** [moneymanagerex/moneymanagerex#6139]
Moneymanagerex crashes and shuts down when deleting a transaction. When started again a window opens asking if you want to use the same database.

<hr> • Version: 1.6.4 64-bit<br>
 • Built: Jun 20 2023 18:27:14<br>
 • db 18 (aes128cbc)<br>
 • git 6b4cc69 (2023-06-20)<br>
<br>
Libs:<br>
 • wxWidgets 3.2.1 (wxMSW 10.0)<br>
 • wxSQLite3 4.9.1 (SQLite 3.40.0)<br>
 • RapidJSON 1.1.0<br>
 • Lua 5.3.5<br>
 • libcurl/7.82.0-DEV Schannel<br>
 • gettext 0.19.8.1<br>
 • apexcharts.js<br>
<br>
Built with:<br>
 • CMake 3.16.2<br>
 • MSBuild 15.9.21.664<br>
 • MSVS15.0<br>
 • MSVSC  19.16.27035.0<br>
 • Windows SDK 10.0.14393.0<br>
<br>
OS:<br>
 • Windows 11 (build 22622), 64-bit edition<br>
 •  English (United Kingdom) (windows-1252)<br>
 • 1920x1080 32bit 96x96ppi

**Comment by n-stein** [comment]
I am also not able to reproduce this. Are you experiencing this every time you delete a transaction, or did this just happen once?

**Comment by johnnyab47** [comment]
This only happens when you right click the transaction and use the drop down menu to delete it. It works correctly if you use the delete button at the bottom of the screen.

**Comment by tactilis** [comment]
> This only happens when you right click the transaction and use the drop down menu to delete it.

You get the crash when trying to right-click delete _any_ transaction or just certain ones?

<br>

Probably not relevant, but what value do you have for `Options | Others | Days to retain deleted transactions`?

Do you get a crash log? Or does MMEX just exit without handling the crash?

**Comment by johnnyab47** [comment]
It crashes when i use delete selected transaction. Retained deleted transactions in options set at 30 days. Yes i get a crash log.

![image](https://github.com/moneymanagerex/moneymanagerex/assets/84288371/968ae96f-30d5-4a03-b55c-272b827f578f)

**Comment by hobbit06** [comment]
I am also experiencing exactly same issue.
![Screenshot 2023-08-05 132405](https://github.com/moneymanagerex/moneymanagerex/assets/141482359/b23e6bba-c962-4759-b222-49e74a0b53da)

**Comment by n-stein** [comment]
Any volunteers to test the 1.7.0 beta against a copy of their database to check if this issue persists in the latest development builds?

**Comment by jerome-duchene** [comment]
I'm volunteer

**Comment by jerome-duchene** [comment]
Exactly the same issue with the 1.7.0 beta 1.
Tested with the portable version with a copy of my db. I have optimize the DB and check the integrity, all is ok.

I run another test, in the contextual menu, I choose "Delete all transactions in the view". In the confirmation popup, I click on Cancel and the app crash also.

I attach 2 dump for this test.
[MoneyManagerEx_dbgrpt-51248-20231117T180028.zip](https://github.com/moneymanagerex/moneymanagerex/files/13394643/MoneyManagerEx_dbgrpt-51248-20231117T180028.zip)
[MoneyManagerEx_dbgrpt-13040-20231117T175643.zip](https://github.com/moneymanagerex/moneymanagerex/files/13394644/MoneyManagerEx_dbgrpt-13040-20231117T175643.zip)

**Comment by n-stein** [comment]
If you go to Tools->Database->Database Check and Debug what is the result of the database integrity check?

**Comment by n-stein** [comment]
>in the contextual menu, I choose "Delete all transactions in the view". In the confirmation popup, I click on Cancel and the app crash also.

This is helpful. If you get to the confirmation popup, click "No" and the app crashes, then the issue is in either the refresh of the checking list or the refresh of the navigation panel.

**Comment by n-stein** [comment]
Still stuck on this one. Can anyone share a sanitized database where this issue is reproducible? Maybe take a copy of your database, delete all transactions (and payees etc. if desired), then see if it crashes when you create a test transaction and attempt to delete it.

**Comment by jerome-duchene** [comment]
hello,
I create a copy of my db, delete all accounts except on bank and cash accounts.
For these 2 accounts, I clear all transaction and... the app doesn't crash when deleting transaction.

I retest with a new copy without clearing it, and again not application crash.

For the testing, I put my tests DB on another folder (c:\temp).

So, for the final, I test on my original DB located in my **_OneDrive folder_** and the **_app crash_** with the same operation.
The "bug" is related when the DB is store on a cloud drive. I suppose that when the cloud agent sync the database after a change, he put a lock on it and MMEX doesn't understand how he can deal with that.
I'm not sure that you can find a workaround for that.

I love having this file on a cloud drive for the synchronisation between my computers.

Since my last test, I had some trouble with the install version so I work with the portable version:
• Version: 1.7.0 64-bit
• Built on: Dec 26 2023 19:10:28
• Database version: 19 (aes128cbc)
• Git commit: b380493 (2023-12-26)

MMEX is using the following support products:
• wxWidgets 3.2.4 (wxMSW 10.0)
• wxSQLite3 4.9.1 (SQLite 3.40.0)
• RapidJSON 1.1.0
• Lua 5.3.5
• libcurl/7.82.0-DEV Schannel
• gettext 0.19.8.1
• apexcharts.js
Built with:
• CMake 3.16.2
• MSBuild 15.9.21.664
• Microsoft Visual Studio 15.0
• Microsoft Visual C++ 19.16.27035.0
• Windows SDK 10.0.14393.0
Running on:
• Windows 11 (build 22631), 64-bit edition
• System Locale: 
• User Interface Language: en_US.windows-1252
• 1920x1200 32-bit 96x96ppi

**Comment by whalley** [comment]
> The "bug" is related when the DB is store on a cloud drive. I suppose that when the cloud agent sync the database after a change, he put a lock on it and MMEX doesn't understand how he can deal with that.

I have always stored my DB on OneDrive with no issues.

**Comment by jerome-duchene** [comment]
My bad, I can reproduce the bug event if I move the database to another drive

**Comment by Ocirasai** [comment]
Posting here as well as in my defect that was closed as a duplicate (#6142).

I'm using the latest stable 1.7.0 released on December 26, 2023.

I created a Test Database and only added one account with one transaction in it.

My steps to reproduce the issue were to:

1. Copy the existing transaction (Using Ctrl-C and then Ctrl-V).
2. Right click on the new transaction and then select Delete > Delete selected transaction...

I've attached the sample database I used along with the crash dump that was created.

Thank you.

[Test.zip](https://github.com/moneymanagerex/moneymanagerex/files/13806262/Test.zip)

**Comment by n-stein** [comment]
Thanks @Ocirasai. I can't reproduce with the attached file using the steps you outlined, this makes me suspect the issue is in the settings db. Can you send us your mmexini.db3 file?

**Comment by Ocirasai** [comment]
Where is that file located?

I can't seem to find it on my Windows 10 system.

**Comment by whalley** [comment]
> Where is that file located?
> 
> I can't seem to find it on my Windows 10 system.

C:\Users\<your username>\AppData\Roaming\MoneyManagerEx

**Comment by Ocirasai** [comment]
Found it.

Had to zip it so I could attach it.

[mmexini.zip](https://github.com/moneymanagerex/moneymanagerex/files/13806583/mmexini.zip)

**Comment by n-stein** [comment]
Hmm, still no success in reproducing this. Any luck @whalley or @vomikan?

**Comment by Ocirasai** [comment]
From the comments above I can see that I am not the only person that has experienced this issue.

Yet, from the change in tags, it looks like it is no longer marked as a bug and tagged as non-reproducible (even though I can reproduce it every time).

Does that mean this issue will not get addressed?

Thank you.

**Comment by whalley** [comment]
It's more that the developers cannot reproduce this. Until we can it is difficult to identify where the issue is.

**Comment by Ocirasai** [comment]
Is there something possibly with Windows 10 (or the Windows platform in general) that might be causing this issue?

Is there a way to get a debug build for 1.7.x that I can try running that might include more detailed output to help understand where it is failing and why?

**Comment by n-stein** [comment]
Here's a portable version of the latest beta (commit https://github.com/moneymanagerex/moneymanagerex/commit/0627e1d243e1ccd23799baa0edf181cb0e1521ed) built with debugging symbols. It's too big to upload to GitHub, so I've put a folder out on my OneDrive. This will produce more verbose debug logs.
[MMEX Beta Debug Builds](https://1drv.ms/f/s!AtNFu4NWfsey0N9AbCfzQUcrIoDdJg?e=WdYXYd)

**Comment by Ocirasai** [comment]
[MMEX.zip](https://github.com/moneymanagerex/moneymanagerex/files/13824771/MMEX.zip)

Zip file containing the test database I used, the database backup that was created when I opened the database, the mmexini.db3 file, and the debug log.

Please let me know if anything else is needed.

**Comment by n-stein** [comment]
Perfect. I updated to the latest commit then made a small tweak based on your stack trace. Can you try the new debug beta build I just uploaded to the same folder?

**Comment by Ocirasai** [comment]
OK - So this time, the behavior mimicked what happens if you hit the delete button at the bottom to delete a transaction rather than crashing. It brings you back out to the Dashboard after deleting the transaction.

I'm not sure if that is intended behavior or not as intuitively I would think it should just delete the transaction and leave you in the account you are working in.

However, I will take this new behavior over the program crashing each time a transaction delete is performed.

**Comment by n-stein** [comment]
> the behavior mimicked what happens if you hit the delete button at the bottom to delete a transaction rather than crashing. It brings you back out to the Dashboard after deleting the transaction.

Interesting. Neither of those things should drop you back to the Dashboard. Whether I use the delete button or right click delete it still stays on the account view for me. Let me take a look at that issue now.

**Comment by n-stein** [comment]
So an identical portable app build executed with the same settings db and transaction db is behaving differently on two different Windows PCs. It must be something within Windows itself is causing the issue. What are your locale, language, and region settings in Windows?

**Comment by Ocirasai** [comment]
![image](https://github.com/moneymanagerex/moneymanagerex/assets/140843655/21efa9ce-0a49-40d4-87b2-15115b09b160)

I'm in the United States, under EST (GMT -5:00).

**Comment by jerome-duchene** [comment]
For my part, it's Belgium:
![image](https://github.com/moneymanagerex/moneymanagerex/assets/17760168/d5f740f5-fde1-43d7-83b2-ea1d961479aa)
![image](https://github.com/moneymanagerex/moneymanagerex/assets/17760168/b8462f39-a5e9-4127-9ac4-02a659601922)

**Comment by n-stein** [comment]
That should at least fix the crash. I will open a new issue for the "Deleting a transaction drops back to dashboard" issue.

**Comment by Ocirasai** [comment]
From what I can see, there are 3 different ways to delete one or more highlighted transactions - Through the menu (Delete > Delete selected transactions), by clicking the Delete button at the bottom of the page, or by pressing the Delete key directly.

Interestingly, when I highlighted a transaction and pressed the [Delete] key directly, I stayed in the account (I was not dropped back to the Dashboard) and there was no crash. I'm not sure if these three operations are handled separately, but if they are, that might be a place to look in the code to see what is different amongst each.
